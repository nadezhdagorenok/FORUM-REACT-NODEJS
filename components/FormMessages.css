.FormMessages{
    align-self: center;
    width: 50vw;
    height: 20vw;
    border: 2px solid cornflowerblue;
    padding: 1vw;
    margin-bottom: 3vw;
}
.FormForum{
    width: 50vw;
    height: 20vw;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
}
.InputBlock{
    align-self: center;
    display: flex;
    justify-content: space-between;
    align-items: center;
    width:100%;
}

.HeaderMessage, .TextMessage{
    font-size: 1.1em;
    align-self: flex-start;
}
.Header-control, .Message-control, .Header-control-active, .Message-control-active{
    width: 40vw;
    align-self: flex-end;
    transition: all 0.30s ease-in-out;
    box-shadow: 0 0 5px rgba(81, 203, 238, 1);
}
.Header-control, .Header-control-active {
    height: 2vw;
}
.Message-control, .Message-control-active {
    height: 7vw;
}
.Header-control-active, .Message-control-active{
    border: 5px solid green;
}
.SendButton{
    width: 7vw;
    height: 3vw;
    font-size: 1.3em;
    background-color: #FF4081;
    border-radius: 5px;
    align-self: flex-end;
    cursor: pointer;

}
