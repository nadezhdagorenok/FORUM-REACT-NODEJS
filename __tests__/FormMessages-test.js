import React from 'react';
import renderer from 'react-test-renderer';

import FormMessages from '../components/FormMessages';

describe('<FormMessages />', () => {
    it('Renders FormMessages', () => {
        const formSubmit = jest.fn();
        const component = renderer.create(
            <FormMessages  cbSendClicked={formSubmit}
                           valueButton='Send'
                           typeButton = 'submit'
            />
        );
        const wrap = component.toJSON();
        expect(wrap).toMatchSnapshot();
    });
});