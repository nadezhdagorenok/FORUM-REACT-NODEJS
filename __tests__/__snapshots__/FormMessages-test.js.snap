// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormMessages /> Renders FormMessages 1`] = `
<div
  className="FormMessages"
>
  <form
    className="FormForum"
  >
    <div
      className="InputBlock"
    >
      <label
        className="HeaderMessage"
        htmlFor="header"
      >
        Header
      </label>
      <input
        autoFocus={true}
        className="Header-control"
        id="header"
        name="header"
        onChange={[Function]}
        placeholder="Type header"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="InputBlock"
    >
      <label
        className="TextMessage"
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        className="Message-control"
        id="message"
        maxLength="140"
        name="message"
        onChange={[Function]}
        placeholder="Type your message"
        rows="7"
        value=""
      />
    </div>
    <button
      className="SendButton"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
